src/multisig/multisig.h

```diff
@@ -28,44 +28,123 @@
 
 #pragma once
 
-#include <vector>
-#include <unordered_map>
 #include "crypto/crypto.h"
 #include "cryptonote_basic/cryptonote_format_utils.h"
 #include "ringct/rctTypes.h"
 
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+
 namespace cryptonote
 {
   struct account_keys;
 
+  struct multisig_account
+  {
+    // an M-of-N multisig participant
+
+    // [M] minimum number of co-signers to sign a message with the aggregate pubkey
+    std::uint32_t m_threshold;
+    // [N] base keys of all participants in the multisig
+    std::vector<crypto::public_key> m_signers;
+
+    // the account's private key shares of the multisig address
+    // TODO: also record which other signers have these privkeys, to enable aggregation signing (instead of round-robin)
+    std::vector<crypto::secret_key> m_privkeys;
+    // a privkey shared by all multisig participants (e.g. private view key)
+    crypto::secret_key m_aggregate_ancillary_privkey;
```

Comment:
My goal was for `multisig_account` to be conceptually isolated from its users. A multisig account has a shared private key (ancillary), and a split private key (for multisig signing).

Reply:
In my super preliminary review, I have a similar comment to this effect too. Perhaps there could be more descriptive names (like `aggregate_view_privkey` instead of `aggregate_ancillary_privkey`, and `base_kex_privkey`/`base_kex_and_signing_privkey` [kinda gross] instead of `base_privkey` etc.) that achieve the same goal. The term `ancillary` relies on more context to know what it actually is, makes it a little harder to keep all the logic of what's happening/which keys are used for what in my head.

I don't have a strong opinion here/don't wanna bikeshed either, whatever you guys think is best is fine by me.

Reply:
What does that mean - it could be use for purposes other than wallet?

Reply:
Theoretically, a multisig account can be used for any miscellaneous purpose. My point is 'separation of concerns' dictates that multisig accounts should not care about/depend on its users. I will think more about how to name things more clearly.